File: 01-collatz/collatz.php
<?php

function collatz($n) {
    $cnt = 0;
    while ($n > 1) {
        $n = ($n%2 ? 3*$n+1 : $n/2);
        $cnt++;
    }
    return $cnt;
}

$max = getenv('MAXN');

$sum = 0;
for ($i = 1; $i <= $max; $i++) {
    $sum += collatz($i);
}

$avg = $sum / $max;
echo "sum=$sum, avg=$avg\n";

File: 01-collatz/collatz.py
import os

def collatz(n):
    cnt = 0
    while n > 1:
        n = 3*n+1 if n%2>0 else n//2
        cnt += 1
    return cnt

n = int(os.getenv('MAXN'))
s = 0
for i in range(1, n+1):
    s += collatz(i)
print("sum=%s, avg=%s" % (s, s/n))
